src/ui/simple_ui/ask_hash.rs

```diff
@@ -55,6 +98,35 @@ pub fn ask_hash(args: &BurnArgs, cf: CompressionFormat) -> anyhow::Result<Option
     Ok(Some(hash_result))
 }
 
+fn find_hash(input: &PathBuf) -> Option<(HashAlg, &str, Vec<u8>)> {
+    for (alg, hash_file) in HASH_FILES {
+        let hash_filepath = input.parent()?.join(hash_file);
+        if let Ok(true) = fs::exists(&hash_filepath) {
+            if let Some(expected_hash) = parse_hashfile(
+                &mut BufReader::new(File::open(hash_filepath).ok()?),
+                input.file_name()?.to_str()?,
+            ) {
+                return Some((alg, hash_file, expected_hash));
+            }
+        }
+    }
+
+    None
+}
+
+fn parse_hashfile(hash_file: &mut impl BufRead, input_file: &str) -> Option<Vec<u8>> {
+    for line in hash_file.lines() {
+        match line.ok()?.split_once(char::is_whitespace) {
+            Some((hash, file)) if file.trim_start() == input_file => {
+                return base16::decode(hash.as_bytes()).ok()
+            }
+            _ => return None,
```

Comment:
why is this a return none? if it's because our file may be invalid shafile probably at least add a comment noting that